subprojects/core/src/main/java/org/gradle/util/internal/NameMatcher.java

```diff
@@ -145,56 +148,46 @@ public String find(String pattern, Collection<String> items) {
         return null;
     }
 
-    private static Pattern getPatternForName(String name) {
-        Pattern boundaryPattern = Pattern.compile("((^|\\p{Punct})\\p{javaLowerCase}+)|(\\p{javaUpperCase}\\p{javaLowerCase}*)");
-        Matcher matcher = boundaryPattern.matcher(name);
-        int pos = 0;
-        StringBuilder builder = new StringBuilder();
-        while (matcher.find()) {
-            String prefix = name.substring(pos, matcher.start());
-            if (prefix.length() > 0) {
-                builder.append(Pattern.quote(prefix));
-            }
-            builder.append(Pattern.quote(matcher.group()));
-            Integer currentJavaVersionMajor = Jvm.current().getJavaVersionMajor();
-            if (currentJavaVersionMajor != null && currentJavaVersionMajor >= 15) {
-                builder.append("[\\p{javaLowerCase}\\p{Digit}]*");
-            } else {
-                // Manually extend character class instead of properties
-                // due to a known issue in regex below Java 15
-                // (https://bugs.openjdk.org/browse/JDK-8239887).
-                // \u00DF-\u00F6(ß-ö) - Covers extended Latin lowercase letters including German
-                //  Eszett and various accented characters.
-                // \u00F8-\u00FF(ø-ÿ) - Covers extended Latin lowercase letters including ø,
-                //  accented vowels, and the thorn.
-                // Note: this approach is not as comprehensive as properties.
-                builder.append("[a-z\\u00DF-\\u00F6\\u00F8-\\u00FF\\p{Digit}]*");
```

Comment:
🤔 How come this is no longer needed?

I ran the test locally with Java 8, 11, 17 and 22. They all pass 🎉 
Do you know which change makes https://bugs.openjdk.org/browse/JDK-8239887 no longer applicable?

Context: #23580

Reply:
It should be `| Pattern.UNICODE_CASE`:
https://github.com/gradle/gradle/blob/e54b94dd0b6732ea92bd5b967c0836630c90ae49/subprojects/core/src/main/java/org/gradle/util/internal/NameMatcher.java#L85